### ISSUE TITLE
Upgrade to Python3

### ISSUE THREAD

**Issue by Seb35** [Gandi/gandi-vm-config#11]
Currently on Debian 10 this package depends on the package `python`, which is `python2` (which reached its end-of-life on January 1, 2020). It should be upgraded to Python3.

On a fresh Debian 10, it is the only package depending on python2.

**Comment by dholl** [comment]
The longer this issue remains unaddressed, the more it is becoming critical.

According to this correspondence dated 2020-09-16, "The python package is now removed in testing":
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=970425

I found this out the hard way when updating packages on a Debian "testing" system, and aptitude wanted to "fix" the issues by removing the gandi-vm-config2 package.

Fortunately I caught it, and pinned these 3 packages at their last version that is depended upon by the ```python``` package (2.7.17-2), thus preventing their automatic upgrade 2.7.18 (which ```python``` does not depend on):
```libpython2-stdlib```
```python2```
```python2-minimal```

Anyhow, if gandi-vm-config doesn't update to python3, I'd hope that switching the ```gandi-vm-config2``` package to depend on ```python2``` instead of ```python``` would ease the transition for everyone in the next Debian stable release.

**Comment by dholl** [comment]
I just downloaded and ran ```apt install ./gandi-hosting-vm2_2.38_all.deb```, so now I'm typing these notes, as I proceed through the remaining dependency issues:

Since the ```python``` package was still at version 2.7.17-2 (and thus less than 2.8), it was still considered a dependency of ```gandi-hosting-vm2 2.38```, and thus ```aptitude``` was still not releasing it even though ```python2``` is installed.  I told ```aptitude``` to just remove ```python```, and then that also released ```python-minimal```.  Then, my system somehow had ```libpython-stdlib``` marked as not-automatically installed, but once I marked it as auto, then ```aptitude``` finished upgrading these python2-related packages from 2.7.17-2 to 2.7.18-2:
	```libpython2-stdlib```
	```python2```
	```python2-minimal```

In conclusion, the happy ending is that the system didn't try to resolve any package problems by removing ```gandi-hosting-vm2```.  I didn't immediately notice any errors after these package updates, nor after rebooting in hopes of trying to provoke problems.

Thank you!  :-)

**Comment by Seb35** [comment]
The PR #26 is a first draft to remove Python3, but only a limited testing was done on it.

I issued an nmu version (non-maintainer upload) on my fork [2.41+nmu1](https://github.com/Seb35/gandi-vm-config/tree/debian/2.41+nmu1), to be packaged with (tested on Debian 11 as root):
```shell
apt-get install build-essential devscripts
git clone -b debian/2.41+nmu1 https://github.com/Seb35/gandi-vm-config
cd gandi-vm-config
dpkg-buildpackage -us -uc
```
The .deb is in the parent directory (`gandi-hosting-vm2_2.41+nmu1_all.deb`). This is provided without any guarantee.

**Comment by svillemot** [comment]
Due to this issue, `gandi-hosting-vm2` is now uninstallable on Debian 12 “bookworm“, since the latter no longer ships Python 2.

So this is a pretty critical issue. I’m now forced to install unsupported Python 2 binaries on my Gandi Cloud server to make it work… Or is `gandi-hosting-vm2` no longer needed/recommended?

**Comment by gdchamal** [comment]
As a quick solution you can remove `gandi-hosting-vm2` package before. This package is mostly needed at first boot to configure correctly the VM.

**Comment by svillemot** [comment]
My impression is that `gandi-hosting-vm2` mounts the virtual disk(s), so I’m not very eager to remove it. But maybe an equivalent solution can be created using entries in `/etc/fstab` or `.mount` systemd units.

**Comment by mmalter-at-gandi** [comment]
2.42 was deployed yesterday for Debian, with python3 support.
